src/AzureAuth/CommandMain.cs

```diff
@@ -382,12 +382,18 @@ public bool PCADisabled()
         private bool ValidateOptions()
         {
             bool validOptions = true;
-            if (string.IsNullOrEmpty(this.authSettings.Resource))
+
+            if (string.IsNullOrEmpty(this.authSettings.Resource) && (this.authSettings.Scopes == null || this.authSettings.Scopes.Count == 0))

```

Comment:
Can we use the `?` operator here like we do in the conditional below?

Reply:
I tried the `?` operator, unfortunately, the answer is `NO`.

For example, when I use 
```csharp
this.authSettings.Scopes?.Count
```
when `this.authSettings.Scopes` is null, this expression `this.authSettings.Scopes?.Count` will be also null value rather than 0.

Reply:
That is to say, if the list is null,
```csharp
this.authSettings.Scopes?.Count == 0
```
will always be false. That's very counterintuitive.

Reply:
In this case - I would expect the line below doing 
```csharp
this.authSettings.Scopes?.Count > 0
```
to have the same potential runtime issue of comparing a `null` to `0`.

You could avoid this in both cases by extracting the value into a local variable above both statements like so:
```csharp
int scopesCount = this.authSettings.Scopes?.Count ?? 0;
```
And then you can clean up both comparisons in each conditional.

Reply:
Good idea. Thank you @kyle-rader.